src/code/z_map_exp.c

```diff
@@ -99,66 +112,75 @@ s32 Map_GetBossAreaIndex(PlayState* play) {
 }
 
 /**
- * Returns true if the current scene is a boss scene, false otherwise.
+ * Checks if the current scene is a boss scene.
+ * @return true if the current scene is a boss scene, false otherwise.
  */
-s32 Map_IsInBossArea(PlayState* play) {
-    if (Map_GetBossAreaIndex(play) == -1) {
+s32 Map_IsInBossScene(PlayState* play) {
+    if (MapExp_TryGetDungeonSceneIndexForBoss(play) == -1) {
         return false;
     }
 
     return true;
 }
 
-s32 D_801BF5A4[] = {
-    SCENE_22DEKUCITY,
-    SCENE_KOEPONARACE,
-    SCENE_F01,
-};
-
-s32 func_8010A238(PlayState* play) {
+/**
+ * Test if the current scene is in a specific list of scenes.
+ * This function may be incomplete.
+ * @return -1 if not in the list, else returns the DungeonSceneIndex for that scene
+ */
+s32 MapExp_TryGetDungeonSceneIndexForUnknownBehavior(PlayState* play) {
+    static s32 sSceneIds[] = {
+        SCENE_22DEKUCITY,
+        SCENE_KOEPONARACE,
+        SCENE_F01,
+    };
     s32 i;
 
-    for (i = 0; i < ARRAY_COUNT(D_801BF5A4); i++) {
-        if (Play_GetOriginalSceneId(play->sceneId) == D_801BF5A4[i]) {
+    for (i = 0; i < ARRAY_COUNT(sSceneIds); i++) {
+        if (Play_GetOriginalSceneId(play->sceneId) == sSceneIds[i]) {
             return i;
         }
     }
 
     return -1;
 }
 
-s32 func_8010A2AC(PlayState* play) {
-    if (func_8010A238(play) == -1) {
+/**
+ * Checks if the current scene is in a specific list of scenes.
+ * The purpose of this grouping of scenes is unknown and is likely incomplete.
+ * @return true if the current scene is in the set, false otherwise.
+ */
+s32 Map_IsInUnknownBehaviorScene(PlayState* play) {
```

Comment:
```suggestion
s32 Map_IsInDungeonOverworldScene(PlayState* play) {
```
Think we could probably just refer to them as `Dungeon Overworld` scenes.